### ISSUE TITLE
Nested map 

### ISSUE THREAD

**Issue by cjohansen** [bhb/expound#27]
I have a case that's working with spec, but failing with expound:

```clj
(require '[clojure.spec.alpha :as s]
         '[expound.alpha :as e])

(s/def :db/valueType #{:db.type/ref})
(s/def :db/cardinality #{:db.cardinality/one :db.cardinality/many})
(s/def :db/unique #{:db.unique/identity})
(s/def ::attribute-schema (s/keys :opt [:db/valueType :db/cardinality :db/unique]))
(s/def ::schema (s/map-of keyword? ::attribute-schema))

(s/explain-str ::schema {:some/value {:db/unique :db.unique/value}})
;;In: [:some/value 1 :db/unique] val: :db.unique/value fails spec: :db/unique at: [1 :db/unique] ;;predicate: #{:db.unique/identity}
;;:clojure.spec.alpha/spec  :user/schema
;;:clojure.spec.alpha/value  #:some{:value #:db{:unique :db.unique/value}}

(e/expound-str ::schema {:some/value {:db/unique :db.unique/value}})
;;AssertionError Assert failed: clojure.lang.LazySeq@8de948aa
;;(every? :expound/in leaf-problems)  expound.alpha/printer-str (alpha.cljc:541)
```

Clojure 1.9.0-alpha19

**Comment by bhb** [comment]
@cjohansen Thanks very much for reporting this. I'll take a look ASAP.

**Comment by cjohansen** [comment]
Awesome. Let me know if you need help or more information.

**Comment by bhb** [comment]
@cjohansen Thanks for reporting this! Would you mind trying out 0.3.1-SNAPSHOT to confirm this works with your specs?

**Comment by cjohansen** [comment]
Works perfectly! Nice work 👍

**Comment by bhb** [comment]
Thanks again for your help on this! I've just released Expound 0.3.1 which includes this fix.
